src/yz_solrq_helper.erl

```diff
@@ -0,0 +1,274 @@
+                ok ->
+                    send_solr_ops(Index, solr_ops(LI, Entries));
+                blown ->
+                    ?ERROR("Fuse Blown: can't currently send solr "
+                           "operations for index ~s", [Index]),
+                    {error, fuse_blown};
+                _ ->
+                    %% fuse table creation is idempotent and occurs on
+                    %% yz_index:add_index/1 on 1st creation or diff-check.
+                    %% We send entries until we can ask again for
+                    %% ok | error, as we wait for the tick.
+                    send_solr_ops(Index, solr_ops(LI, Entries))
+            end
+    end.
+
+%% @doc Build the SOLR query
+-spec solr_ops(logical_idx(), solr_entries()) -> solr_ops().
+solr_ops(LI, Entries) ->
+    lists:reverse(
+      lists:foldl(
+        fun({BKey, Obj0, Reason0, P, ShortPL, Hash}, Ops) ->
+            {Bucket, _} = BKey,
+            BProps = riak_core_bucket:get_bucket(Bucket),
+            Obj = yz_kv:maybe_merge_siblings(BProps, Obj0),
+            ObjValues = riak_object:get_values(Obj),
+            Reason = get_reason_action(Reason0),
+            case {Reason, ObjValues} of
+                {delete, _} ->
+                    [{delete, yz_solr:encode_delete({bkey, BKey})} | Ops];
+                {_, [notfound]} ->
+                    [{delete, yz_solr:encode_delete({bkey, BKey})} | Ops];
+                _ ->
+                    LFPN = yz_cover:logical_partition(LI, element(1, ShortPL)),
+                    LP = yz_cover:logical_partition(LI, P),
+                    Docs = yz_doc:make_docs(Obj, Hash, ?INT_TO_BIN(LFPN),
+                                            ?INT_TO_BIN(LP)),
+                    AddOps = yz_doc:adding_docs(Docs),
+                    DeleteOps = yz_kv:delete_operation(BProps, Obj, Docs, BKey,
+                                                       LP),
+                    %% List will be reversed, so make sure deletes happen
+                    %% before adds
+                    lists:append([[{add, yz_solr:encode_doc(Doc)} || Doc <- AddOps],
+                                  [{delete, yz_solr:encode_delete(DeleteOp)} ||
+                                      DeleteOp <- DeleteOps],
+                                  Ops])
+            end
+        end, [], Entries)).
+
+-spec send_solr_ops(index_name(), solr_ops()) -> ok.
+send_solr_ops(Index, Ops) ->
+    try
+        T1 = os:timestamp(),
+        ok = yz_solr:index_batch(Index, Ops),
+        yz_stat:index_end(Index, length(Ops), ?YZ_TIME_ELAPSED(T1))
+    catch _:Err ->
+            yz_stat:index_fail(),
+            Trace = erlang:get_stacktrace(),
+            ?ERROR("batch for index ~s failed - ~p\n with operations: ~p : ~p",
+                   [Index, Err, Ops, Trace]),
+            case Err of
+                {_, badrequest, _} ->
+                    send_solr_single_ops(Index, Ops),
+                    ok;
```

Comment:
@fadushin made a good point here. send_solr_ops needs to filter out the bad_request entries before tree update, passing in the correct entrylist. I'll create a sep PR off this branch for review.